### PR TITLE
Lint common JS files and test specs

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Lints our common JS files that are shared in many pages.
- Lints our JS test specs and adds them to the ESLint grunt task.

## Bugzilla
N/A

## Testing
I've given this some manual testing, and changes seem pretty safe. I guess the main thing is to make sure any renamed functions aren't also called from anywhere else that I may have missed. I've done a check and think all is ok, but a second sweep would be good.

## Checklist
- [ ] Requires l10n changes.
- [ ] Related functional & integration tests passing.

### CODE REVIEW

File: tests/unit/spec/base/mozilla-accordion.js [hunk 1]
```diff
@@ -14,33 +15,33 @@ describe('mozilla-accordion.js', function () {
...
+                    '<div data-accordion-role="tabpanel"><p>Section 1 contents.</p></div>' +
+                '</section>' +
+                '<section>' +
+                    '<h2 data-accordion-role="tab">Section 2</h2>' +
+                    '<div data-accordion-role="tabpanel"><p>Section 2 contents.</p></div>' +
+                '</section>' +
+                '<section>' +
+                    '<h2 data-accordion-role="tab">Section 3</h2>' +
+                    '<div data-accordion-role="tabpanel"><p>Section 3 contents.</p></div>' +
+                '</section>' +
```
Comment (schalkneethling) [inline tests/unit/spec/base/mozilla-accordion.js]: Wait so, maybe I am not reading this correctly but, seeing you are now creating a string by concatenation, does this still need the array and the call to `join`?

Comment (alexgibson) [inline tests/unit/spec/base/mozilla-accordion.js]: I just forgot to remove the join most likely

Comment (schalkneethling) [inline tests/unit/spec/base/mozilla-accordion.js]: Aha, so I am not going crazy ;) There is another instance of it in a file further down.

File: tests/unit/spec/firefox/australis/browser-tour.js [hunk 2]
```diff
@@ -25,71 +26,71 @@ describe('browser-tour.js', function() {
...
+            '<div id="ui-tour-mask" tabindex="-1">' +
+                '<div class="mask-inner">' +
+                    '<div class="logo"></div>' +
+                    '<div class="stage">' +
+                        '<h1>Mask title</h1>' +
+                        '<p>Mask sub heading</p>' +
+                    '</div>' +
+                '</div>' +
+            '</div>'
         ].join();
```
Comment (schalkneethling) [inline tests/unit/spec/firefox/australis/browser-tour.js]: over here.

**Comment by schalkneethling** [comment]
Browsed around the site in Fx, Chrome and IE8, all works as expected. Ran Grunt tests, all pass. So, other than the two items mentioned r+ :100:

**Comment by alexgibson** [comment]
@schalkneethling updated thanks

**Pull request merged**